mindsdb_native/libs/backends/lightwood.py

```diff
@@ -395,37 +392,54 @@ def predict(self, mode='predict', ignore_columns=None, all_mixers=False):
         else:
             run_df = df
 
-        predictions = self.predictor.predict(when_data=run_df)
+        if self.transaction.lmd['tss']['is_timeseries']:
+            run_df, _ = self._create_timeseries_df(run_df)
+
+        predictions = {}
+        predictions[0] = self.predictor.predict(when_data=run_df)
+
+        # additional predictions for timeseries
+        timesteps = self.transaction.lmd.get('nr_predictions', 1)
+        if self.transaction.lmd['tss']['is_timeseries']:
+            for t in range(1, timesteps):
+                displaced_df = copy.deepcopy(df)
+                for target in self.predictor._output_columns:
+                    displaced_df[target] = predictions[t-1][target]['predictions']
+                # if self.transaction.lmd['tss']['use_previous_target']:
+                displaced_df, _ = self._create_timeseries_df(displaced_df)
```

Comment:
The logic here seems incorrect, the way to do it (I'd think) is something like:

1. create a new row that contains all the values from the row used to predict + the value for the `target` set to the prediction
2. append that row to `df` (or rather to a copy of it, but only make this copy onceA)
3. `displaced_df, _ = self._create_timeseries_df(copy_of_df_with_row_appended)`
4. Rinse and repeat (appending to the same df